### PR TITLE
Add retry mechanism when registering to the server

### PR DESCRIPTION
**Pull request by poedja**

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -31,6 +34,7 @@ func CmdAgent(ctx *cli.Context) {
 	})
 }
 
+
```
Comment (giosakti) [inline commands.go]: please remove this unnecessary extra whitespace

File: commands.go [hunk 2]
```diff
@@ -11,10 +10,14 @@ import (
 	"github.com/pathfinder-cm/pathfinder-agent/agent"
 	"github.com/pathfinder-cm/pathfinder-agent/daemon"
 	"github.com/pathfinder-cm/pathfinder-go-client/pfclient"
+	"github.com/pathfinder-cm/pathfinder-agent/util"
 	log "github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
 )
 
+//"errors"
```
Comment (giosakti) [inline commands.go]: please remove this unnecessary extra comments and linebreaks

File: commands.go [hunk 3]
```diff
@@ -56,10 +60,18 @@ func runAgent() {
...
-	// Self Register
-	ok, _ := pfclient.Register(hostname, ipaddress)
-	if !ok {
-		panic(errors.New("Cannot register to pathfinder server, please check your configuration."))
+	for {
+		log.WithFields(log.Fields{}).Warn("Trying to register to pathfinder server...")
+		ok, _ := pfclient.Register(hostname,ipaddress)
+		
+		if !ok {
+			log.Error("Cannot register to pathfinder server, please check your configuration")
```
Comment (giosakti) [inline commands.go]: can we add `Retrying in x seconds...`
and calculate the delay before this line

Comment (poedja) [reply commands.go]: Added the log message

File: commands.go (hunk 1 above)
Comment (poedja) [reply commands.go]: Done  f46ae53

File: commands.go (hunk 2 above)
Comment (poedja) [reply commands.go]: Done. f46ae53

File: commands.go (hunk 3 above)
Comment (poedja) [reply commands.go]: Done f46ae53

**Pull request merged**